rules/xcodeproj.bzl

```diff
@@ -40,9 +42,13 @@ def _xcodeproj_aspect_impl(target, ctx):
             test_env_vars = tuple(env_key_value_pairs.items())
             commandlines_args = getattr(ctx.rule.attr, "args", [])
             test_commandline_args = tuple(commandlines_args)
+            test_host_target = getattr(ctx.rule.attr, "test_host", None)
+            if test_host_target and test_host_target.label.package != "rules/test_host_app":
+                test_host_appname = test_host_target.label.name
```

Comment:
is this the name of the app, or the name of the target that produces the app?

Reply:
Looks like we might want the `bundle_name` from the `ios_application` target, rather than just its labels name. We might need to dig in and get it via a provider from that target

Reply:
It is the name of the bazel target that produces the app.

In the xcodeproject, the xcode test target should point to the xcode target name of the test host application, which should match the name of the bazel target that produces the app.

Does that all sound right?

Reply:
That does sound right, but it seems this variable is also used for the _app_ name (when setting the TESTHOST build setting), which can be different from the target name